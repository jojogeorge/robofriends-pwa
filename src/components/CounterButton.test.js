import {shallow} from 'enzyme'
import React from 'react';
import renderer from 'react-test-renderer'
import CounterButton from './CounterButton'

describe('CounterButton Component',()=> {


    it('expect CardList compoent to be renedered', ()=>{
        expect(shallow(<CounterButton/>).length).toEqual(1);
        });
        
    it('expect CardList compoent to be renedered', ()=>{
        
        const tree = renderer.create(<CounterButton />).toJSON();
        expect(tree).toMatchSnapshot();
    
    });
    it('correctly increments counter', ()=>{
        
        const wrapper = shallow(<CounterButton/>);
        expect(wrapper.state()).toEqual({count:0});
        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({count:1});

        //expect(wrapper.props().color).toEqual('red');
    
    });

})

