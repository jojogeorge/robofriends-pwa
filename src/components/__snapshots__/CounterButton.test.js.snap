// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component expect CardList compoent to be renedered 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  count:
  0
</button>
`;
